chore(docs): Adds two key steps on building ecommerce sites with Stripe tutorial

urielhdz opened the pull request:
## Description

After following the tutorial on [how tu build ecommerce sites with Stripe](https://www.gatsbyjs.org/tutorial/ecommerce-tutorial) I noticed that two important steps were missing from the tutorial:

1. **SKUs need to have a name attribute for them to be used on the client only Checkout integration**: I added the steps to add the name through the Stripes dashboard, although the attribute can also be set through the API, I think we could fix this issue by just adding the requirement as an annotation instead of explaining the whole process, but since adding the attribute is not that straightforward I added the steps to the tutorial.
2. **Checkout client-side only integration needs to be enabled to use the redirectToCheckout function**: This is key since the guide is based on this specific functionality, since this issue is solved only through the Stripes dashboard, I think that is safe to keep the section on how to enable the integration.

urielhdz commented:
Seems like the Stripe's interface changed for me 🙃, now SKUs require a name once they are created. I don't know how to tackle these scenarios in which the UI seems to differ per account.

urielhdz commented:
I removed the section on how to set a name for SKUs since the latest UI requires a name by default.

Now this PR limits to the second point, which reminds you that the Checkout client-side only integration needs to be enabled.

LekoArts commented on docs/tutorial/ecommerce-tutorial/index.md:
````diff
@@ -171,6 +171,10 @@ const Checkout = class extends React.Component {
 export default Checkout
 ```
 
+Be sure to enable the Checkout client-only integration on your Stripe dashboard to use the redirectToCheckout integration, you may do that in this url: https://dashboard.stripe.com/account/checkout/settings
````
```suggestion
Be sure to enable the "Checkout client-only integration" on your Stripe dashboard to be able to use the redirectToCheckout integration. You can activate the integration in your [Stripe settings](https://dashboard.stripe.com/account/checkout/settings).
```

Let's use a nicer link to the settings and clarify the wording a bit.

urielhdz replied:
Thank you @LekoArts I just pushed the requested changes.

urielhdz commented:
Upon further investigation, It seems like the SKUs interface changes after the "Checkout client-only integration" is enabled.

muescha commented on docs/tutorial/ecommerce-tutorial/index.md:
````diff
@@ -171,6 +171,10 @@ const Checkout = class extends React.Component {
 export default Checkout
 ```
 
+Be sure to enable the "Checkout client-only integration" on your Stripe dashboard to be able to use the redirectToCheckout integration. You can activate the integration in your [Stripe settings](https://dashboard.stripe.com/account/checkout/settings).
````
```suggestion
Be sure to enable the "Checkout client-only integration" on your Stripe dashboard to be able to use the `redirectToCheckout` integration. You can activate the integration in your [Stripe settings](https://dashboard.stripe.com/account/checkout/settings).
```

muescha replied:
You use double in this line and others - and I can not comment on lines not in PR changes:

Can you also change from single quotes to double quotes in line 117

> `'Viewing test data'`

muescha commented on docs/tutorial/ecommerce-tutorial/index.md:
```diff
@@ -171,6 +171,10 @@ const Checkout = class extends React.Component {
 
+Be sure to enable the "Checkout client-only integration" on your Stripe dashboard to be able to use the redirectToCheckout integration. You can activate the integration in your [Stripe settings](https://dashboard.stripe.com/account/checkout/settings).
+
+![Stripe control to enable the Checkout client-side only integration highlighted](stripe-checkout-clientside-functionlality.png)
```
I filename is also this typo

```suggestion
![Stripe control to enable the Checkout client-side only integration highlighted](stripe-checkout-clientside-functionality.png)
```

urielhdz commented:
Thanks to both @muescha and @LekoArts for your improvements.

Given how the SKUs interface changes depending on your Stripe's settings, and the fact that you may be using the API and not the dashboard, I rewrote the **Create a product and SKU**  to mention that your SKUs need to have a name for them to be used on this integration, and because there was a reference of Stripe working with any Backend component.

I also created a new account to follow the process from the start and I added instruction on how to properly setup your account to use this integration, this includes enabling the 'Checkout client-only integration' and setting a name for your account.

I also changed some pronouns from 'we' to 'you' that were left on the tutorial to adhere to #18284.

Additionally, I was wondering if components should be changed from class based to functional? In this particular example we would need to change both the tutorial and the provided code from the demo.

Thanks again!

muescha commented:
i think updating it to functional/hooks would be a nice improvement.

urielhdz commented:
> i think updating it to functional/hooks would be a nice improvement.

I already have something working, let me push it to the PR

muescha commented:
wow great!

urielhdz commented:
> wow great!

I am going to fork the example to test the code and the changes, there are multiple components that need to be changed, I don't know if it is best to open a new PR to make changes easier to review or do you agree on pushing them here? @muescha

urielhdz commented:
Let me know if there is anything missing for this to be merged 😅

LekoArts commented:
Let's wait for @gatsbyjs/learning to comment on that, too!

urielhdz commented:
Thank you @laurieontech ! I added your suggestions to the PR!

marcysutton commented on docs/tutorial/ecommerce-tutorial/index.md:
````diff
@@ -282,7 +294,7 @@ module.exports = {
 ```
 
-To retrieve your SKUs from your Stripe account you will need to provide your secret API key. This key needs to kept secret and must never be shared on the frontend or on GitHub. Therefore we need to set an environment variable to store the secret key. You can read more about the usage of env variables in Gatsby [here](https://www.gatsbyjs.org/docs/environment-variables/).
+To retrieve your SKUs from your Stripe account you will need to provide your secret API key. This key needs to be kept secret and must never be shared on the frontend or on GitHub. Therefore you need to set an environment variable to store the secret key. You can read more about the usage of env variables in Gatsby [here](https://www.gatsbyjs.org/docs/environment-variables/).
````
We usually add more specific text for URLs to improve accessibility, so this would be good to fix here (it also changes it to use a relative URL):
```suggestion
To retrieve your SKUs from your Stripe account you will need to provide your secret API key. This key needs to be kept secret and must never be shared on the frontend or on GitHub. Therefore you need to set an environment variable to store the secret key. You can read more about the usage of env variables in the [Gatsby docs](/docs/environment-variables/).
```

marcysutton commented on docs/tutorial/ecommerce-tutorial/index.md:
```diff
@@ -112,9 +124,9 @@ If you're selling a simple product, like an eBook for example, you can create a
 #### Create a product and SKU
 
-For Stripe Checkout to work without any backend component, you need to create a product listing in the Stripe Dashboard. This is required for Stripe to validate that the request coming from the frontend is legitimate and to charge the right amount for the selected product/SKU. To set this up, simply follow the steps in the [Stripe docs](https://stripe.com/docs/payments/checkout#configure).
+To sell your products, first you need to create them on Stripe using the [Stripe Dashboard](https://dashboard.stripe.com/products) or the [Stripe API](https://stripe.com/docs/api/products/create). This is required for Stripe to validate that the request coming from the frontend is legitimate and to charge the right amount for the selected product/SKU. Stripe requires every SKU used with Stripe Ch ...
```
```suggestion
To sell your products, first you need to create them on Stripe using the [Stripe Dashboard](https://dashboard.stripe.com/products) or the [Stripe API](https://stripe.com/docs/api/products/create). This is required for Stripe to validate that the request coming from the frontend is legitimate and to charge the right amount for the selected product/SKU. Stripe requires every SKU used with Stripe Checkout to have a name: be sure to add one to all of your SKUs.
```

marcysutton commented on docs/tutorial/ecommerce-tutorial/index.md:
```diff
@@ -102,6 +102,18 @@ You have 2 keys in both test mode and production mode:
+
+![Stripe control to enable the Checkout client-side only integration highlighted](stripe-checkout-clientside-functionality.png)
+
+> 💡 This change will also modify the interface that Stripe provides to administrate your products, keep this in mind in case you have previously used this tool. If you have never used the product administrator, you don't need to worry.
```
```suggestion
> 💡 This change will also modify the interface that Stripe provides to administer your products: keep this in mind in case you have previously used this tool. If you have never used the product administrator, you don't need to worry.
```

urielhdz commented:
Thanks, @marcysutton! I just applied your suggestions, let me know if is there anything else I can do for this to be merged

urielhdz commented:
Hello! Let me know if is there anything that needs to be done for this to be merged and I'd be happy to work on it! 

@marcysutton @laurieontech 

Thanks!

gillkyle commented:
Hey @urielhdz, I know this PR has been up and you want to get it over the finish line, I'm going to do my best to look over this today or tomorrow so we can merge it in. Just want to let you know so you're not in the dark 🙂
